### ISSUE TITLE
Hang on using Windows's Switch User

### ISSUE THREAD

**Issue by thefeel** [luke-jr/bfgminer#103]
BFGminer 2.6.4 hangs when using switch user on a Windows 7 x64 machine.
The interface stops responding and updating, and no shares are submitted even though the gpus continue running the kernels.
I have tried using different versions of cgminer all the way to 2.0.4, they all show the same behavior.

Other miners like phoenix 1.7.5 and 2.0, poclbm, guiminer continue mining after switching user various times.

System:
Windows 7 x64 ultimate
Catalyst 12.6, APP 2.7
AMD A6-3650, Radeon 7950 and Radeon 5850

**Comment by luke-jr** [comment]
- Hanging only occurs when using curses UI (ie, --text-only works fine)
- ADL returns -1 for all values when other user is active; this means temp/fanspeed/etc will not work!

Backtrace after user switching (with 1a3785a):

```
Thread 21 (Thread 2364.0x117c):
#0  0x76f2000d in ntdll!LdrFindResource_U ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76faf896 in ntdll!RtlQueryTimeZoneInformation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x7d60b3f6 in ?? ()
#3  0x00000000 in ?? ()

Thread 20 (Thread 2364.0xe40):
#0  0x76f2f8b1 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f2f8b1 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c00a91 in WaitForSingleObjectEx ()
   from C:\Windows\syswow64\KernelBase.dll
#3  0x00000120 in ?? ()
#4  0x00000000 in ?? ()

Thread 19 (Thread 2364.0x964):
#0  0x76f2f8b1 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f2f8b1 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c00a91 in WaitForSingleObjectEx ()
   from C:\Windows\syswow64\KernelBase.dll
#3  0x00000120 in ?? ()
#4  0x00000000 in ?? ()

Thread 18 (Thread 2364.0x950):
#0  0x76f31f26 in ntdll!LdrQueryProcessModuleInformation ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f31f26 in ntdll!LdrQueryProcessModuleInformation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x76f63352 in ntdll!RtlCreateTagHeap () from C:\Windows\SysWOW64\ntdll.dll
#3  0x00000a10 in ?? ()
#4  0x09a9fedc in ?? ()
#5  0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#6  0x0083ba80 in ?? ()
#7  0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#8  0x0083ba80 in ?? ()
#9  0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#10 0x76f63e45 in ntdll!RtlSidIsHigherLevel ()
   from C:\Windows\SysWOW64\ntdll.dll
#11 0x0083ba80 in ?? ()
#12 0x00000000 in ?? ()

Thread 17 (Thread 2364.0x12cc):
#0  0x76f31f26 in ntdll!LdrQueryProcessModuleInformation ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f31f26 in ntdll!LdrQueryProcessModuleInformation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x76f63352 in ntdll!RtlCreateTagHeap () from C:\Windows\SysWOW64\ntdll.dll
#3  0x00000a10 in ?? ()
#4  0x0985fedc in ?? ()
#5  0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#6  0x0083ba80 in ?? ()
#7  0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#8  0x0083ba80 in ?? ()
#9  0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#10 0x76f63e45 in ntdll!RtlSidIsHigherLevel ()
   from C:\Windows\SysWOW64\ntdll.dll
#11 0x0083ba80 in ?? ()
#12 0x00000000 in ?? ()

Thread 16 (Thread 2364.0x1210):
#0  0x76f31f26 in ntdll!LdrQueryProcessModuleInformation ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f31f26 in ntdll!LdrQueryProcessModuleInformation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x76f63352 in ntdll!RtlCreateTagHeap () from C:\Windows\SysWOW64\ntdll.dll
#3  0x000009f4 in ?? ()
#4  0x093dfedc in ?? ()
#5  0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#6  0x0083c710 in ?? ()
#7  0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#8  0x0083c710 in ?? ()
#9  0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#10 0x76f63e45 in ntdll!RtlSidIsHigherLevel ()
   from C:\Windows\SysWOW64\ntdll.dll
#11 0x0083c710 in ?? ()
#12 0x00000000 in ?? ()

Thread 15 (Thread 2364.0x1150):
#0  0x76f2fd71 in ntdll!RtlFindSetBits () from C:\Windows\SysWOW64\ntdll.dll
#1  0x74c031bb in SleepEx () from C:\Windows\syswow64\KernelBase.dll
#2  0x00000000 in ?? ()

Thread 14 (Thread 2364.0xac0):
#0  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c00bdd in WaitForMultipleObjectsEx ()
   from C:\Windows\syswow64\KernelBase.dll
#3  0x00000002 in ?? ()
#4  0x091dfb78 in ?? ()
#5  0x749d1a2c in KERNEL32!GetVolumePathNamesForVolumeNameA ()
   from C:\Windows\syswow64\kernel32.dll
#6  0x091dfb78 in ?? ()
#7  0x749d4208 in KERNEL32!CheckForReadOnlyResource ()
   from C:\Windows\syswow64\kernel32.dll
#8  0x00000002 in ?? ()
#9  0x7efde000 in ?? ()
#10 0x6248b9ec in pthreadGC2!pthreadCancelableTimedWait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#11 0x6248c6b6 in sem_timedwait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#12 0x6248cf8b in pthread_cond_wait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#13 0x00415505 in tq_pop (tq=0x517660, abstime=0x0) at util.c:649
#14 0x00425830 in reinit_gpu (userdata=0x501a20) at driver-opencl.c:1364
#15 0x6248b4c1 in pthread_create ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#16 0x74cd1287 in msvcrt!_itow_s () from C:\Windows\syswow64\msvcrt.dll
#17 0x74cd1328 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#18 0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#19 0x0052e150 in ?? ()
#20 0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#21 0x0052e150 in ?? ()
#22 0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#23 0x74cd12e5 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#24 0x00000000 in ?? ()

Thread 13 (Thread 2364.0xf98):
#0  0x76f2f8b1 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f2f8b1 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c00a91 in WaitForSingleObjectEx ()
   from C:\Windows\syswow64\KernelBase.dll
#3  0x00000120 in ?? ()
#4  0x00000000 in ?? ()

Thread 12 (Thread 2364.0x1278):
#0  0x76f2fd71 in ntdll!RtlFindSetBits () from C:\Windows\SysWOW64\ntdll.dll
#1  0x74c031bb in SleepEx () from C:\Windows\syswow64\KernelBase.dll
#2  0x00000001 in ?? ()
#3  0x08ddfe04 in ?? ()
#4  0x0040194b in nanosleep (req=0x8ddfe98, rem=0x8ddfe90) at compat.h:22
#5  0x00401a8c in sleep (secs=30) at compat.h:51
#6  0x0040f013 in watchpool_thread (userdata=0x0) at miner.c:5131
#7  0x6248b4c1 in pthread_create ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#8  0x74cd1287 in msvcrt!_itow_s () from C:\Windows\syswow64\msvcrt.dll
#9  0x74cd1328 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#10 0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#11 0x0052dbc0 in ?? ()
#12 0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#13 0x0052dbc0 in ?? ()
#14 0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#15 0x74cd12e5 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#16 0x00000000 in ?? ()

Thread 11 (Thread 2364.0x119c):
#0  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c00bdd in WaitForMultipleObjectsEx ()
   from C:\Windows\syswow64\KernelBase.dll
#3  0x00000002 in ?? ()
#4  0x08bdfbc8 in ?? ()
#5  0x749d1a2c in KERNEL32!GetVolumePathNamesForVolumeNameA ()
   from C:\Windows\syswow64\kernel32.dll
#6  0x08bdfbc8 in ?? ()
#7  0x749d4208 in KERNEL32!CheckForReadOnlyResource ()
   from C:\Windows\syswow64\kernel32.dll
#8  0x00000002 in ?? ()
#9  0x7efde000 in ?? ()
#10 0x6248b9ec in pthreadGC2!pthreadCancelableTimedWait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#11 0x6248c6b6 in sem_timedwait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#12 0x6248cf8b in pthread_cond_wait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#13 0x00415505 in tq_pop (tq=0x517560, abstime=0x0) at util.c:649
#14 0x0040dcc8 in mt_disable (mythr=0x501890, thr_id=1, api=0x4584c0)
    at miner.c:4643
#15 0x0040e619 in miner_thread (userdata=0x501890) at miner.c:4818
#16 0x6248b4c1 in pthread_create ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#17 0x74cd1287 in msvcrt!_itow_s () from C:\Windows\syswow64\msvcrt.dll
#18 0x74cd1328 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#19 0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#20 0x00528cf8 in ?? ()
#21 0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#22 0x00528cf8 in ?? ()
#23 0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#24 0x74cd12e5 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#25 0x00000000 in ?? ()

Thread 10 (Thread 2364.0x1314):
#0  0x76f2f8b1 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f2f8b1 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c00a91 in WaitForSingleObjectEx ()
   from C:\Windows\syswow64\KernelBase.dll
#3  0x00002c54 in ?? ()
#4  0x00000000 in ?? ()

Thread 9 (Thread 2364.0x1048):
#0  0x76f2ff31 in ntdll!RtlImpersonateSelfEx ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f2ff31 in ntdll!RtlImpersonateSelfEx ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c0344c in SwitchToThread () from C:\Windows\syswow64\KernelBase.dll
#3  0x69714818 in clGetSamplerInfo () from C:\Windows\system32\amdocl.dll
#4  0x6970cf19 in clGetSamplerInfo () from C:\Windows\system32\amdocl.dll
#5  0x696fcca3 in clEnqueueReadBuffer () from C:\Windows\system32\amdocl.dll
#6  0x72fc1e3e in clEnqueueReadBuffer () from C:\Windows\system32\OpenCL.dll
#7  0x00426b45 in opencl_scanhash (thr=0x501840, work=0x517248,
    max_nonce=1941962751) at driver-opencl.c:1842
#8  0x0040e222 in miner_thread (userdata=0x501840) at miner.c:4747
#9  0x6248b4c1 in pthread_create ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#10 0x74cd1287 in msvcrt!_itow_s () from C:\Windows\syswow64\msvcrt.dll
#11 0x74cd1328 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#12 0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#13 0x00517248 in ?? ()
#14 0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#15 0x00517248 in ?? ()
#16 0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#17 0x74cd12e5 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#18 0x00000000 in ?? ()

Thread 8 (Thread 2364.0x1250):
#0  0x76f2ff31 in ntdll!RtlImpersonateSelfEx ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f2ff31 in ntdll!RtlImpersonateSelfEx ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c0344c in SwitchToThread () from C:\Windows\syswow64\KernelBase.dll
#3  0x69714818 in clGetSamplerInfo () from C:\Windows\system32\amdocl.dll
#4  0x6970d1c5 in clGetSamplerInfo () from C:\Windows\system32\amdocl.dll
#5  0x6970d56a in clGetSamplerInfo () from C:\Windows\system32\amdocl.dll
#6  0x69705edb in clGetSamplerInfo () from C:\Windows\system32\amdocl.dll
#7  0x69708ba6 in clGetSamplerInfo () from C:\Windows\system32\amdocl.dll
#8  0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#9  0x02c838d0 in ?? ()
#10 0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#11 0x02c838d0 in ?? ()
#12 0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#13 0x69708b70 in clGetSamplerInfo () from C:\Windows\system32\amdocl.dll
#14 0x00000000 in ?? ()

Thread 7 (Thread 2364.0xe54):
#0  0x76f2f8b1 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f2f8b1 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x72dd17cd in ?? () from C:\Windows\System32\mswsock.dll
#3  0x72dd6d30 in ?? () from C:\Windows\System32\mswsock.dll
#4  0x74966a28 in WSAAsyncGetHostByAddr () from C:\Windows\syswow64\ws2_32.dll
#5  0x00000911 in ?? ()
#6  0x057df814 in ?? ()
#7  0x708238f3 in Curl_socket_check ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\libcurl-4.dll
#8  0x057df918 in ?? ()
#9  0x00000910 in ?? ()
#10 0x000000e8 in ?? ()
#11 0x00501f80 in ?? ()
Backtrace stopped: previous frame inner to this frame (corrupt stack?)

Thread 6 (Thread 2364.0x1174):
#0  0x76f2f939 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f2f939 in ntdll!RtlUpdateClonedSRWLock ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x72dd635c in ?? () from C:\Windows\System32\mswsock.dll
#3  0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#4  0x72dd64b3 in ?? () from C:\Windows\System32\mswsock.dll
#5  0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#6  0x008398e8 in ?? ()
#7  0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#8  0x72dd62ee in ?? () from C:\Windows\System32\mswsock.dll
#9  0x00000000 in ?? ()

Thread 5 (Thread 2364.0x13d8):
#0  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x76f62f51 in ntdll!RtlWeaklyEnumerateEntryHashTable ()
   from C:\Windows\SysWOW64\ntdll.dll
#3  0x00000003 in ?? ()
#4  0x008615e8 in ?? ()
#5  0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#6  0x00000000 in ?? ()

Thread 4 (Thread 2364.0xc6c):
#0  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c00bdd in WaitForMultipleObjectsEx ()
   from C:\Windows\syswow64\KernelBase.dll
#3  0x00000002 in ?? ()
#4  0x04b5fc78 in ?? ()
#5  0x749d1a2c in KERNEL32!GetVolumePathNamesForVolumeNameA ()
   from C:\Windows\syswow64\kernel32.dll
#6  0x04b5fc78 in ?? ()
#7  0x749d4208 in KERNEL32!CheckForReadOnlyResource ()
   from C:\Windows\syswow64\kernel32.dll
#8  0x00000002 in ?? ()
#9  0x7efde000 in ?? ()
#10 0x6248b9ec in pthreadGC2!pthreadCancelableTimedWait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#11 0x6248c6b6 in sem_timedwait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#12 0x6248cf8b in pthread_cond_wait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#13 0x00415505 in tq_pop (tq=0xaff368, abstime=0x0) at util.c:649
#14 0x00409aeb in stage_thread (userdata=0x501930) at miner.c:3277
#15 0x6248b4c1 in pthread_create ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#16 0x74cd1287 in msvcrt!_itow_s () from C:\Windows\syswow64\msvcrt.dll
#17 0x74cd1328 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#18 0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#19 0x00501530 in ?? ()
#20 0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#21 0x00501530 in ?? ()
#22 0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#23 0x74cd12e5 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#24 0x00000000 in ?? ()

Thread 3 (Thread 2364.0xbb4):
#0  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c00bdd in WaitForMultipleObjectsEx ()
   from C:\Windows\syswow64\KernelBase.dll
#3  0x00000002 in ?? ()
#4  0x0495fc88 in ?? ()
#5  0x749d1a2c in KERNEL32!GetVolumePathNamesForVolumeNameA ()
   from C:\Windows\syswow64\kernel32.dll
#6  0x0495fc88 in ?? ()
#7  0x749d4208 in KERNEL32!CheckForReadOnlyResource ()
   from C:\Windows\syswow64\kernel32.dll
#8  0x00000002 in ?? ()
#9  0x7efde000 in ?? ()
#10 0x6248b9ec in pthreadGC2!pthreadCancelableTimedWait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#11 0x6248c6b6 in sem_timedwait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#12 0x6248cf8b in pthread_cond_wait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#13 0x00415505 in tq_pop (tq=0xaf9e90, abstime=0x0) at util.c:649
#14 0x0040bfd3 in workio_thread (userdata=0x5018e0) at miner.c:4034
#15 0x6248b4c1 in pthread_create ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#16 0x74cd1287 in msvcrt!_itow_s () from C:\Windows\syswow64\msvcrt.dll
#17 0x74cd1328 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#18 0x749d339a in KERNEL32!BaseCleanupAppcompatCacheSupport ()
   from C:\Windows\syswow64\kernel32.dll
#19 0x00508578 in ?? ()
#20 0x76f49ef2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#21 0x00508578 in ?? ()
#22 0x76f49ec5 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SysWOW64\ntdll.dll
#23 0x74cd12e5 in msvcrt!_endthreadex () from C:\Windows\syswow64\msvcrt.dll
#24 0x00000000 in ?? ()

Thread 2 (Thread 2364.0x1204):
#0  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c00bdd in WaitForMultipleObjectsEx ()
   from C:\Windows\syswow64\KernelBase.dll
#3  0x00000002 in ?? ()
#4  0x0295fe48 in ?? ()
#5  0x749d1a2c in KERNEL32!GetVolumePathNamesForVolumeNameA ()
   from C:\Windows\syswow64\kernel32.dll
#6  0x0295fe48 in ?? ()
#7  0x749d4208 in KERNEL32!CheckForReadOnlyResource ()
   from C:\Windows\syswow64\kernel32.dll
#8  0x00000002 in ?? ()
#9  0x7efde000 in ?? ()
#10 0x6b608554 in windows_clock_gettime_threaded@4 ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\libusb-1.0.dll
#11 0xf6e8c116 in ?? ()
#12 0x74d62bd0 in msvcrt!_iob () from C:\Windows\syswow64\msvcrt.dll
#13 0x00af2f70 in ?? ()
Backtrace stopped: previous frame inner to this frame (corrupt stack?)

Thread 1 (Thread 2364.0xeec):
#0  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#1  0x76f3013d in ntdll!RtlEnableEarlyCriticalSectionEventCreation ()
   from C:\Windows\SysWOW64\ntdll.dll
#2  0x74c00bdd in WaitForMultipleObjectsEx ()
   from C:\Windows\syswow64\KernelBase.dll
#3  0x00000002 in ?? ()
#4  0x0028fa98 in ?? ()
#5  0x749d1a2c in KERNEL32!GetVolumePathNamesForVolumeNameA ()
   from C:\Windows\syswow64\kernel32.dll
#6  0x0028fa98 in ?? ()
#7  0x749d4208 in KERNEL32!CheckForReadOnlyResource ()
   from C:\Windows\syswow64\kernel32.dll
#8  0x00000002 in ?? ()
#9  0x7efde000 in ?? ()
#10 0x6248bd41 in pthreadGC2!pthreadCancelableWait ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#11 0x6248c237 in pthread_join ()
   from c:\Users\luke-jr\Documents\bfgminer-2.6.4-win32\bfgminer-2.6.4-win32\pthreadGC2.dll
#12 0x0041366f in main (argc=1, argv=0xaf2e38) at miner.c:6281
```

**Comment by thefeel** [comment]
--text-only works, but bfgminer still complains about adl, and restarts all threads thinking the gpus crashed. At least it doesn't crash anymore.

**Comment by luke-jr** [comment]
Hanging is caused by a recursive attempt to lock the console:

```
\-- _curses_active_locked
|   \-- _lock_curses
|   |   \-- _mutex_lock
| ...
\-- _curses_print_devstatus
|   \-- _get_opencl_statline_before
|   | ...
|   |   \-- _gpu_fanpercent
|   |   | ...
|   |   |   \-- _applog
|   |   |   |   \-- _vapplog
|   |   |   |   |   \-- _log_generic
|   |   |   |   |   |   \-- _my_log_curses
|   |   |   |   |   |   |   \-- _log_curses_only
|   |   |   |   |   |   |   |   \-- _curses_active_locked
|   |   |   |   |   |   |   |   |   \-- _lock_curses
|   |   |   |   |   |   |   |   |   |   \-- _mutex_lock
```

Please open another issue for the "thinking the gpus crashed" part, since it's not directly related to the one initially reported here.

**Comment by luke-jr** [comment]
Since this is pretty much just Con's domain, I've [provided him with the information](https://bitcointalk.org/index.php?topic=28402.msg1111945#msg1111945) in hopes he will fix it. If not, I'll take another look at doing it myself.

**Comment by luke-jr** [comment]
[04:07] &lt;conman> oh a fix for the user s witch?
[04:07] &lt;conman> who cares
[04:07] &lt;conman> fucking windows pos

Guess I get to poke at it...

**Comment by luke-jr** [comment]
Fixed in ccf77ca

**Comment by luke-jr** [comment]
Combined with --no-restart, this should make it usable at least. I highly advise against using this with the overclocking or fanspeed controls enabled until the ADL issue is fixed (please open a new issue for that).
